atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -494,6 +491,35 @@ Set<CassandraServer> validateNewHostsTopologiesAndMaybeAddToPool(
         return validatedServersToAdd;
     }
 
+    @VisibleForTesting
+    Set<CassandraServer> tryGettingNewHostsWithDifferentTopologyOrInvalidateNewContainersAndThrow(
+            Map<CassandraServer, CassandraServerOrigin> serversToAdd,
+            Map<CassandraServer, CassandraClientPoolingContainer> allContainers) {
+        try {
+            // Max duration is one minute as we expect the cluster to have recovered by then due to gossip.
+            return cassandraTopologyValidator.getNewHostsWithInconsistentTopologiesAndRetry(
+                    serversToAdd, allContainers, Duration.ofSeconds(5), Duration.ofMinutes(1));
+        } catch (Throwable t) {
+            EntryStream.of(allContainers)
+                    .filterKeys(serversToAdd::containsKey)
+                    .values()
+                    .forEach(CassandraClientPoolImpl::tryShuttingDownCassandraClientPoolingContainer);
+            log.warn("Failed to get new hosts with inconsistent topologies.", t);
+            throw t;
+        }
```

Comment:
Lets not let an exception for a container.shutdown here to interfere with other containers' closure